app/Files/TemporaryFile.php

```diff
@@ -0,0 +1,43 @@
+<?php
```

Comment:
explained somewhere inside.   
1. upload file into temp folder. queue up for processing
2. user cancel upload - >delete temp file
3. user click submit form -> move temp file into actual.